### PR TITLE
PneumoCaT wrapper

### PR DESCRIPTION
**Pull request by DarianHole**
PneumoCaT wrapper finished.

PneumoCaT works on Pneumococcal Capsular Typing and the program is found here: https://github.com/phe-bioinformatics/PneumoCaT

Cheetah code makes sure input names are the same and won't break the program (it needs files named *_1.fastq, *_2.fastq or *_R1.fastq, *_R2.fastq)

Only spot that I know of that could be improved more is dealing with Gzipped filenames. Wrapper works fine with current structure when a gzipped file is input with the correct formatting. It slightly messes up the name on the output file when the formatting is off.

exs.
- test_R1.fastq.gz will run through the program as `test`
- test.fastq.gz will runn through the program as `test_fastq`

### CODE REVIEW

File: tools/pneumocat/.shed.yml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+categories: [Adrian's Wants]
```
[the remark this answers is not in the document]
Comment (DarianHole) [reply tools/pneumocat/.shed.yml]: Woops I missed changing that, now its Variant Analysis

File: tools/pneumocat/pneumocat-3.xml [hunk 2]
```diff
@@ -0,0 +1,81 @@
+<tool id="pneumocat"
+      name="PneumoCaT"
+      version="0.1.0">
```
[the remark this answers is not in the document]
Comment (DarianHole) [reply tools/pneumocat/pneumocat-3.xml]: Correct, done

**Comment by ericenns** [comment]
@DarianHole can you look at the tests it failed linting

**Comment by DarianHole** [comment]
@ericenns Its WIP as I need to figure out a way to write tests still that don't include the 2 250Mb test files that come with the data

File: tools/pneumocat/pneumocat.xml [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    </macros>
+    <expand macro="requirements" />
+    <command detect_errors="exit_code"><![CDATA[
+    
+    #import os
+
+    #if $input.type == "paired"
+        #set $for_input = "{}.fastq".format(os.path.splitext($input.forward.name)[0])
+        #set $rev_input = "{}.fastq".format(os.path.splitext($input.reverse.name)[0])
+        ln -s  "$input.forward" ./$for_input &&
```
Comment (bgruening) [inline tools/pneumocat/pneumocat.xml]: please use single-quotes instead of double-quotes

File: tools/pneumocat/pneumocat.xml [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+    #else
+        #set $for_input = "{}.fastq".format(os.path.splitext($input.fastq_collection.forward.name)[0])
+        #set $rev_input = "{}.fastq".format(os.path.splitext($input.fastq_collection.reverse.name)[0])
+        ln -s "$input.fastq_collection.forward" $for_input &&
+        ln -s "$input.fastq_collection.reverse" $rev_input &&
+
+    #end if
+
+    PneumoCaT.py -1 "$for_input" -2 "$rev_input" -o outputs --threads "\${GALAXY_SLOTS:-1}" --cleanup
```
Comment (bgruening) [inline tools/pneumocat/pneumocat.xml]: please use single-quotes, please see here for more details: https://galaxy-iuc-standards.readthedocs.io/en/latest/best_practices/tool_xml.html#command-formatting

File: tools/pneumocat/pneumocat.xml [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+                <param name="reverse"
+                    type="data" format="fastq,fastqsanger,fastq.gz,fastqsanger.gz"
+                    optional="false"
+                    label="Reverse reads (FASTQ)"
+                    help="File format must match the Forward FASTQ file"
+                />
+            </when>
+            <when value="paired_collection">
+                <param name="fastq_collection"
+                    type="data_collection" format="fastq,fastqsanger"
```
Comment (bgruening) [inline tools/pneumocat/pneumocat.xml]: why not gz here as well?

Comment (DarianHole) [reply tools/pneumocat/pneumocat.xml]: Woops I missed that, I'll be sure to add it in and then add a test for that too potentially

**Comment by DarianHole** [comment]
Still needs:
- Paired fastq collection test
- Help Section
- Double check everything works as expected

**Pull request merged**